.github/workflows/cicd.yml

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+# This is the top-level workflow that runs on each pull request and push to main.
```

Comment:
Why do we need to run it on a push to main?  (Add answer in the file comment.)  It should have already run on the pull request being merged to main, no?

Reply:
The answer is on line 10. When we do a squash commit to main from a PR, it creates a new unique commit. Codecov.io requires a commit as a baseline to compare against, but in the case of a squash the PR isn't the same commit as the one made to main.